adam-core/src/main/scala/org/bdgenomics/adam/rich/RichGenotype.scala

```diff
@@ -1,112 +0,0 @@
-/**
- * Licensed to Big Data Genomics (BDG) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The BDG licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.bdgenomics.adam.rich
-
-import org.bdgenomics.formats.avro.{ GenotypeType, GenotypeAllele, Genotype }
-import scala.collection.JavaConversions._
-
-object RichGenotype {
-  implicit def genotypeToRichGenotype(g: Genotype) = new RichGenotype(g)
-  implicit def richGenotypeToGenotype(g: RichGenotype) = g.genotype
-}
-
-class RichGenotype(val genotype: Genotype) {
-
-  def ploidy: Int = genotype.getAlleles.size
-
-  def getType: GenotypeType = {
```

Comment:
There is so much difficulty explaining what the different types mean I don't find `GenotypeType` very useful.

Hail defines even more genotype flags

genotype field | definition
------------ | -------------
`isHomRef` | true if this call is 0/0 
`isHet` | true if this call is heterozygous 
`isHetRef` | true if this call is 0/k with k>0 
`isHetNonRef` | true if this call is j/k with j>0 
`isHomVar` | true if this call is j/j with j>0 
`isCalledNonRef` | true if either isHet or isHomVar is true 
`isCalled` | true if the genotype is not ./. 
`isNotCalled` | true if the genotype is ./. 

and still doesn't distinguish partial no-calls.